File: StudentManagementNode/models/Subject.js
import mongoose from 'mongoose';

const subjectSchema = new mongoose.Schema({
  subjectCode: { type: String, required: true, unique: true },  // e.g. "CS101"
  subjectName: { type: String, required: true },
  credits: { type: Number, required: true },
  department: { type: String, required: true }
}, { timestamps: true });

const Subject = mongoose.model('Subject', subjectSchema);
export default Subject;

File: StudentMnagement/src/main/java/com/adobe/StudentManagement/model/Subject.java
package com.adobe.StudentManagement.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "subjects")
public class Subject {

    @Id
    private String id;

    private String subjectCode;
    private String subjectName;
    private Integer credits;
    private String department;
}
